onellm/utils/file_validator.py

```
@@ -0,0 +1,393 @@
+            max_size = DEFAULT_MAX_FILE_SIZE
+        if allowed_extensions is None:
+            allowed_extensions = DEFAULT_ALLOWED_EXTENSIONS
+        
+        # Check for directory traversal attempts BEFORE resolving
+        # This catches patterns like "../../../etc/passwd"
+        # Normalize separators for cross-platform compatibility
+        normalized_path = file_path.replace("\\", "/")
+        if ".." in normalized_path:
+            raise InvalidRequestError(
+                f"Directory traversal detected in path: {file_path}"
+            )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)